javers-persistence-sql/src/test/java/org/javers/repository/sql/cases/Case542CustomSchemaMaager.java

```diff
@@ -0,0 +1,60 @@
+package org.javers.repository.sql.cases;
+
+import org.fest.assertions.api.Assertions;
+import org.javers.core.Javers;
+import org.javers.core.JaversBuilder;
+import org.javers.core.metamodel.annotation.Entity;
+import org.javers.core.metamodel.annotation.Id;
+import org.javers.core.metamodel.object.CdoSnapshot;
+import org.javers.repository.sql.H2RepositoryFactory;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.Serializable;
+
+/**
+ * see https://github.com/javers/javers/issues/249
+ * @author bartosz.walacik
+ */
+public class Case542CustomSchemaMaager {
+
+    public static abstract class AbstractEntity<T extends Serializable> {
+
+        @Id
+        protected T id;
+        protected T value;
+
+        public AbstractEntity(T id, T value) {
+            this.id = id;
+            this.value = value;
+        }
+    }
+
+    @Entity
+    public class Account extends AbstractEntity<String> {
+        public Account(String id, String value) {
+            super(id, value);
+        }
+    }
+
+    @Before
+    public void shouldCreateSchema(){
+        JaversBuilder.javers().registerJaversRepository(H2RepositoryFactory.create()).build();
+    }
+
+    @Test
+    public void shouldCommitEntityWithSerializableId() {
+        //given
+        Javers javers = JaversBuilder.javers()
+                .registerJaversRepository(H2RepositoryFactory.createWithCustomSchemaManager()).build();
```

Comment:
Since CustomSchemaManager does nothing, what is the purpose of this test? What feature are you testing here?

Reply:
even if custom schema manager does not create tables it still write on them, indeed data is written via javers commit. CustomSchemaManager use another change I introduced that is the ability to instance TableNameProvider with an optional instead of using SqlRepositoryConfiguration that has a package visibility